lib/backend/grpcclient/grpc.go

```diff
@@ -43,81 +66,206 @@ func GetName() string {
 // keep this here to test interface conformance
 var _ backend.Backend = &Backend{}
 
-// New returns new instance of grpc-powered backend
+// New returns new instance of gRPC-powered backend.
+//
+// The supplied context is used to cancel or expire the pending connection to
+// the gRPC server. Once New is returned, the cancellation and expiration of ctx
+// will be noop.
 func New(ctx context.Context, params backend.Params) (*Backend, error) {
 	l := log.WithFields(log.Fields{trace.Component: GetName()})
 
 	l.Infof("Initializing backend %v.", GetName())
 	defer l.Debug("Backend initialization complete.")
 
-	return &Backend{}, nil
+	var cfg *Config
+	if err := utils.ObjectToStruct(params, &cfg); err != nil {
+		return nil, trace.BadParameter("gRPC: configuration is invalid: %v", err)
+	}
+
+	if err := cfg.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// CA + client certs for mTLS
+	caCert, err := ioutil.ReadFile(cfg.ServerCA)
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	cert, err := tls.LoadX509KeyPair(cfg.ClientCert, cfg.ClientKey)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// TODO(michaelmcallister): There are some tunables that we may need to expose
+	// or set such as:
+	// - InitialConnWindowSize
+	// - InitialWindowSize
+	// - ReadBufferSize
+	// - WriteBuffersize
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
```

Comment:
I hadn't intended to support this. It would be handy for tests, but even then creating an ephemeral keypair for mTLS in unit tests isn't very hard to do.

Happy to revisit this if there is demand for it though